demos/text-generation/llm.js

```diff
@@ -0,0 +1,321 @@
+                    path: externalFile,
+                },
+            ],
+        };
+
+        const locationType = this.provider == "webnn" ? "ml-tensor" : "gpu-buffer";
+        switch (this.provider) {
+            case "webnn":
+            case "webgpu":
+                // Bind kv cache outputs to ml-tensor or gpu-buffer
+                sessionOptions.preferredOutputLocation = {};
+                for (let i = 0; i < this.numLayers; ++i) {
+                    sessionOptions.preferredOutputLocation[`present.${i}.key`] = locationType;
+                    sessionOptions.preferredOutputLocation[`present.${i}.value`] = locationType;
+                }
+                break;
+            case "wasm":
+                sessionOptions.preferredOutputLocation = "cpu";
+                break;
+        }
+
+        if (verbose) {
+            sessionOptions.logSeverityLevel = 0;
+            sessionOptions.logVerbosityLevel = 0;
+        }
+
+        if (this.provider == "webnn") {
+            sessionOptions.freeDimensionOverrides = {
+                batch_size: 1,
+                sequence_length: this.maxLength,
+                total_sequence_length: this.maxLength,
+                past_sequence_length: this.maxLength,
+            };
+        }
+
+        let progressBarLabel = $("#p-bar-label");
+        log("Create session for prefill process");
+        console.log("Create session 1 with option: ");
+        console.log({ ...sessionOptions });
+        console.log(this.sess1);
+        this.sess1 = await ort.InferenceSession.create(modelBytes, sessionOptions);
+        updateOnnxCompileProgress(10);
+        updateLoadProgress(onnxFetchProgress + onnxDataFetchProgress + onnxCompileProgress + onnxDataCompileProgress);
+        updateProgressBar(loadProgress.toFixed(2));
+        progressBarLabel.innerHTML = `Prefill session created · ${loadProgress.toFixed(2)}%`;
+
+        log("Prefill session created");
+        if (this.provider == "webnn") {
+            // decode process
+            sessionOptions.freeDimensionOverrides = {
+                batch_size: 1,
+                sequence_length: 1,
+                total_sequence_length: this.maxLength,
+                past_sequence_length: this.maxLength,
+            };
+            log("Create session for decode process");
+            console.log("Create session 2 with option: ");
+            console.log({ ...sessionOptions });
+            this.sess2 = await ort.InferenceSession.create(modelBytes, sessionOptions);
+            log("Decode process session created");
+        }
+
+        updateOnnxDataCompileProgress(10);
+        updateLoadProgress(onnxFetchProgress + onnxDataFetchProgress + onnxCompileProgress + onnxDataCompileProgress);
+        updateProgressBar(loadProgress.toFixed(2));
+        progressBarLabel.innerHTML = `Session for decode created · ${loadProgress.toFixed(2)}%`;
+
+        updateProgressBar(100.0);
+        progressBarLabel.innerHTML = `100%`;
```

Comment:
```suggestion
        progressBarLabel.innerHTML = '100%';
```
Any reason for the backquotes? Is any special formatting happening here?